.github/workflows/cd-infrastructure.yml

```diff
@@ -27,4 +28,30 @@ jobs:
       azure_ad_client_secret: ${{ secrets.AZURE_AD_CLIENT_SECRET }}
       azure_subscription_id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
       azure_ad_tenant_id: ${{ secrets.AZURE_AD_TENANT_ID }}
-      azure_storage_account_key: ${{ secrets.AZURE_STORAGE_ACCOUNT_KEY }}
\ No newline at end of file
+      azure_storage_account_key: ${{ secrets.AZURE_STORAGE_ACCOUNT_KEY }}
+  Dev_deployment:
+    if: github.event.pull_request.merged == true
+    uses: ./.github/workflows/reusable-terraform-deployment.yml
+    with:
+      gh_environment: Development
+      working_directory_path: infrastructure
+      environment_name: dev
+    secrets:
+      azure_ad_client_id: ${{ secrets.AZURE_AD_CLIENT_ID }}
+      azure_ad_client_secret: ${{ secrets.AZURE_AD_CLIENT_SECRET }}
+      azure_subscription_id: ${{ secrets.AZURE_SUBSCRIPTION_ID }}
+      azure_ad_tenant_id: ${{ secrets.AZURE_AD_TENANT_ID }}
+      azure_storage_account_key: ${{ secrets.AZURE_STORAGE_ACCOUNT_KEY }}
+  Prod_deployment:
```

Comment:
If possible, I think it would be useful looking into adding some sort of approval mechanism here as we would not want to deploy to prod every time we make a merge to main

Reply:
Agreed - @adelechurchill and I tinkered with Environments and using protection rules to do precisely this.  That wouldn't sit in this file - it's environmental configuration for the defined environments in this repo.  We should ensure we capture this with a story to determine who is in the group who have the power to push to production.

Reply:
Definitely agreed, I think all of us should be able to allow deployment to go ahead, but having that manual step is really important. Would be great to put it between `terraform plan` and `terraform apply` so we have the opportunity to cancel deployment if we see something is wrong in the plan stage.

What do you think?

Reply:
Since prod isn't being used by anyone I would say maybe for the sake of pragmatism we can add the manual intervention step later in the project. For now, let's ship to prod?  Shout if you're even feeling slightly wary about this

Reply:
@decodingahmed really like the idea for having the approval step between plan and apply. Happy with shipping it straight to prod, but for me at least would be nice to play around and understand how the approval part works of GH devops as I've never seen in action before to see what we can achieve with it

Reply:
@yahya130 as Guy states this was covered in our Github Actions spike demo - it is easy to implement but I would agree with @guypritchard that this should be captured by a separate story for full team discussion